docs/docs/mdx/migrate-remark-to-mdx.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Migrating Remark to MDX
+---
+
+For people who already have an existing blog using gatsby-transformer-remark, but want to use MDX, you can swap out the remark transformer plugin with gatsby-plugin-mdx and touch little code otherwise.
+
+## Prerequisites
+
+- An existing site that builds pages using `gatsby-transformer-remark` ([gatsby-starter-blog](https://github.com/gatsbyjs/gatsby-starter-blog) will be used as a reference in this guide).
+
+## Adding in gatsby-plugin-mdx
+
+Add gatsby-plugin-mdx (and its peer dependencies) to your package.json and remove gatsby-transformer-remark.
+
+```shell
+npm install @mdx-js/mdx @mdx-js/react gatsby-plugin-mdx
+npm remove gatsby-transformer-remark
+```
+
+## Replacing gatsby-transformer-remark with gatsby-plugin-mdx
+
+In your gatsby-config.js file, replace gatsby-transformer-remark with gatsby-plugin-mdx. All sub-plugins of gatsby-transformer-remark can still work with gatsby-plugin-mdx by updating the plugins option to gatsbyRemarkPlugins.
```

Comment:
I worry about saying all here. There is at least one I'm aware of that has been known to clash. Maybe saying that you can use remark plugins with mdx without the all piece?